game/Simulation.cpp

```diff
@@ -69,36 +69,32 @@ void Simulation::GetNextEvolution()
 	World* currentWorld = m_world;
 	World* nextWorld = new World(currentWorld->GetSize());
 
-	for (Player* player : m_players)
+	for (int i = 0; i < currentWorld->GetSize(); i++)
 	{
-		Coordinate currentLoc = player->GetLocation();
+		for (int j = 0; j < currentWorld->GetSize(); j++)
+		{
+			Coordinate currentLoc(i, j);
 
-		int numNeighbors = currentWorld->CountNeighbours(currentLoc);
+			int numNeighbors = currentWorld->CountNeighbours(currentLoc);
+			PlayerType newType = player->GetPlayerType();
 
-		PlayerType newType = player->GetPlayerType();
-		if (player->GetPlayerType() == PlayerType::Alive)
-		{
-			if (numNeighbors <= 1 || numNeighbors >= 4)
+			if (player->GetPlayerType() == PlayerType::Alive)
 			{
-				newType = PlayerType::Dead;
+				if (numNeighbors <= 1 || numNeighbors >= 4)
+				{
+					newType = PlayerType::Dead;
+				}
 			}
-		}
-		else if (player->GetPlayerType() == PlayerType::Dead)
-		{
-			if (numNeighbors == 3)
+			else if (player->GetPlayerType() == PlayerType::Dead)
 			{
-				newType = PlayerType::Alive;
+				if (numNeighbors == 3)
+				{
+					newType = PlayerType::Alive;
+				}
 			}
-		}
 
-		player->SetPlayertype(newType);
-
-		Coordinate newLoc = currentLoc;
-		while (!currentWorld->MovePlayer(*player, newLoc))
-		{
-			newLoc = Coordinate(rand() % currentWorld->GetSize(), rand() % currentWorld->GetSize());
+			nextWorld->SetPlayertype(currentLoc, newType);
```

Comment:
This causes an syntax error because its `SetPlayerType` and not `SetPlayertype` (the T is big). C++ is case sensitive